scripts/detectNotUsingGitPush1by1.fsx

```diff
@@ -563,37 +563,441 @@ let prCommits =
+        "downloads_url": "https://api.github.com/repos/realmarv/conventions/downloads",
+        "issues_url": "https://api.github.com/repos/realmarv/conventions/issues{/number}",
+        "pulls_url": "https://api.github.com/repos/realmarv/conventions/pulls{/number}",
+        "milestones_url": "https://api.github.com/repos/realmarv/conventions/milestones{/number}",
+        "notifications_url": "https://api.github.com/repos/realmarv/conventions/notifications{?since,all,participating}",
+        "labels_url": "https://api.github.com/repos/realmarv/conventions/labels{/name}",
+        "releases_url": "https://api.github.com/repos/realmarv/conventions/releases{/id}",
+        "deployments_url": "https://api.github.com/repos/realmarv/conventions/deployments"
+      }
+    }
+  ]
+}    
+""">
 
-        let json = (client.GetStringAsync url).Result
+prCommits
+|> Seq.iter(fun commit ->
+    use client = new HttpClient()
+    client.DefaultRequestHeaders.Accept.Clear()
 
-        json.Contains "\"check_suites\":[]"
+    client.DefaultRequestHeaders.Accept.Add(
+        MediaTypeWithQualityHeaderValue "application/vnd.github+json"
     )
-    |> Seq.contains true
 
-if notUsingGitPush1by1 then
-    let errMsg =
+    client.DefaultRequestHeaders.Add("User-Agent", ".NET App")
+    client.DefaultRequestHeaders.Add("X-GitHub-Api-Version", "2022-11-28")
+
+    let url =
         sprintf
-            "Please push the commits one by one; using this script is recommended:%s%s"
-            Environment.NewLine
-            "https://github.com/nblockchain/conventions/blob/master/scripts/gitPush1by1.fsx"
+            "https://api.github.com/repos/%s/commits/%s/check-suites"
+            gitRepo
+            commit
+
+    let checkSuitsJsonString = (client.GetStringAsync url).Result
+
+    if checkSuitsJsonString.Contains "\"check_suites\":[]" then
+        let errMsg =
+            sprintf
+                "Please push the commits one by one; using this script is recommended:%s%s"
+                Environment.NewLine
+                "https://github.com/nblockchain/conventions/blob/master/scripts/gitPush1by1.fsx"
+
+        Console.Error.WriteLine errMsg
+        Environment.Exit 1
+
+    let checkSuitsParsedJson = checkSuitsType.Parse checkSuitsJsonString
+
+    let commitMessageTitle =
+        (checkSuitsParsedJson.CheckSuites.[0]
+            .HeadCommit
+            .Message)
+            .Split(
+            "\n"
+        ).[0]
+
+    let commitHash = checkSuitsParsedJson.CheckSuites.[0].HeadSha
+    let status = checkSuitsParsedJson.CheckSuites.[0].Status
+    let conclusion = checkSuitsParsedJson.CheckSuites.[0].Conclusion
+
+    if status = "completed" then
+        if commitMessageTitle.Contains "failing test" && conclusion = "success" then
+            Console.Error.WriteLine
+                $"The CI status of the commit \"{commitHash}\" must be `failure`, since it's adding a failing test."
```

Comment:
@realmarv I never asked you to do this (next time you think of new requirements of a task, ask me first please, as it's already explained in the handbook*); in fact it is wrong, because the commit msg could say "fix failing test"

Reply:
Removed it